guardrails/guard.py

```diff
@@ -235,8 +235,15 @@ def configure(
             self._set_tracer(tracer)
         self._configure_telemtry(allow_metrics_collection)
 
-    def _set_num_reasks(self, num_reasks: Optional[int] = 1):
-        self._num_reasks = num_reasks
+    def _set_num_reasks(self, num_reasks: Optional[int]):
```

Comment:
im no lint/typing expert but surprised theres not a version of pyright that wants Optional[int] = None?

Reply:
ok guess not. lgtm

Reply:
tbh I quite detest this function signature.  I could do `_set_num_reasks(self, num_reasks: int = 1)`.

Reply:
Yeah, I like num_reasks: Optional[int] = None much more.  Good suggestion.